keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/ResourceEvent.kt

```diff
@@ -385,14 +385,14 @@ enum class ResourceCheckErrorOrigin {
   @JsonProperty("unknown") UNKNOWN;
 
   companion object {
-    val log: Logger by lazy { LoggerFactory.getLogger(javaClass) }
-    fun fromException(exceptionType: Class<out SpinnakerException>) =
-      when {
-        UserException::class.java.isAssignableFrom(exceptionType) -> USER
-        SystemException::class.java.isAssignableFrom(exceptionType) -> SYSTEM
+    val log: Logger by lazy { LoggerFactory.getLogger(ResourceCheckErrorOrigin.javaClass) }
+    fun fromException(exception: SpinnakerException) =
+      when (exception) {
+        is UserException -> USER
+        is SystemException -> SYSTEM
         else -> UNKNOWN.also {
-          log.error("All keel exceptions should inherit from UserException or SystemException (got ${exceptionType.name}). " +
-            "This is a bug.")
+          log.debug("All keel exceptions should inherit from UserException or SystemException (got ${exception.javaClass.name}). " +
+            "This is a probably a bug.")
```

Comment:
"Probably" because, although it's certainly a bug for new events thrown after the introduction of this check, this code path is also used when instantiating old events from the database.